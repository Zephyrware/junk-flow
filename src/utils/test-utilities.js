import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export const rendersCorrect = component => {
  const wrapper = shallow(component);
  expect(wrapper.length).toBe(1);
};

export const matchesSnapshot = component => {
  const wrapper = shallow(component);
  expect(toJson(wrapper)).toMatchSnapshot();
};
