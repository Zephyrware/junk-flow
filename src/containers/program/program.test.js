// @jest
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import asProgram from './program';
import { rendersCorrect, matchesSnapshot } from '../../utils/test-utilities';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);
const store = mockStore();

class TestComponent extends React.Component {
  render() {
    return <div />;
  }
}

const DecoratedComponent = asProgram({ allowMultipleInstances: false })(
  TestComponent,
);
const getMinComponent = () => <DecoratedComponent store={store} />;

it('AsProgram HOC renders to snapshot correctly', () => {
  matchesSnapshot(getMinComponent());
});

it('AsProgram HOC renders correctly', () => {
  rendersCorrect(getMinComponent());
});

it('AsProgram passed props successfully through', () => {
  const wrapper = shallow(<DecoratedComponent store={store} />);
  console.log(wrapper.props());
  expect(wrapper.props().closeProgramByWindowId).toBeDefined();
  expect(wrapper.props().closeProgramsByProgramId).toBeDefined();
  expect(wrapper.props().openProgram).toBeDefined();
});
