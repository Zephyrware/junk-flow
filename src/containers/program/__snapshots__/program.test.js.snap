// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsProgram HOC renders to snapshot correctly 1`] = `
<WrapperComponent
  closeProgramByWindowId={[Function]}
  closeProgramsByProgramId={[Function]}
  openProgram={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
